common/src/main/java/com/wynntils/wynn/handleditems/WynnItem.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wynn.handleditems;
+
+import com.wynntils.handlers.item.ItemAnnotation;
+import java.util.HashMap;
+import java.util.Map;
+
+public class WynnItem implements ItemAnnotation {
+    private Map<Class<?>, Object> cache = new HashMap<>();
+    private boolean searched = false;
+
+    public <T> T getCached(Class<T> clazz) {
```

Comment:
What is this used for?

Reply:
(And do we GC this..?)

Reply:
It is GC:d when the ItemStack containing this annotation is GC:ed. That's fine for us. They will be GC:ed when they no longer are present in the inventory or in an open chest.

Reply:
It is used to store computationally expensive stuff that is calculated from this WynnItem. Like Tooltip (not yet implemented), and TextOverlay (implemented). So it's free for features etc "downstream" to use to store stuff that is derived from the WynnItem, but that you do not want to repeatedly recalculate. Since the class is the key to the map, you can always put your "own" storage in the object.

Reply:
The "searched" flag is to signal when searching for items in the bank.